tensorboard/webapp/feature_flag/views/feature_flag_modal_trigger_container_test.ts

```diff
@@ -0,0 +1,102 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {NO_ERRORS_SCHEMA} from '@angular/core';
+import {TestBed} from '@angular/core/testing';
+import {
+  MatDialogModule,
+  MatDialogRef,
+  MAT_DIALOG_DATA,
+} from '@angular/material/dialog';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {Action, Store} from '@ngrx/store';
+import {MockStore, provideMockStore} from '@ngrx/store/testing';
+import {Observable, Subscriber} from 'rxjs';
+import {State} from '../../app_state';
+import {
+  getDefaultFeatureFlags,
+  getOverriddenFeatureFlags,
+  getShowFlagsEnabled,
+} from '../store/feature_flag_selectors';
+import {FeatureFlags} from '../types';
+import {FeatureFlagModalTriggerContainer} from './feature_flag_modal_trigger_container';
+import {FeatureFlagPageContainer} from './feature_flag_page_container';
+
+class MatDialogMock {
```

Comment:
https://material.angular.io/cdk/test-harnesses/overview

Sometime in the last 6 months we decided we should use test harnesses when appropriate. This seems like a possible great candidate, given that their first example is about using a harness for MatDialog. Would you mind trying to use that?